ctapipe/io/hessioeventsource.py

```diff
@@ -226,6 +226,7 @@ def _build_telescope_description(self, file, tel_id):
         num_tiles = file.get_mirror_number(tel_id)
         cam_rot = file.get_camera_rotation_angle(tel_id)
         num_mirrors = file.get_mirror_number(tel_id)
+        sampling_rate = u.Quantity(1 / file.get_time_slice(tel_id), u.GHz)
```

Comment:
Sorry, working on a temporary desk top, could you tell me what get_time_slice returns here? Is it the length of the readout? I just worry that if it is just the number of samples then it would only convert to GHz assuming that the sampling rate itself is 1GHz (i.e. does the function take into account the FADC_MHz and FADC_SUM_BINS parameters from sim_telarray)?

Reply:
From pyhessio:
```
    def get_time_slice(self, telescope_id):
        """
        Returns Width of readout time slice (i.e. one sample) [ns].
         If telescope_id is not valid return 0.
        Parameters
        ----------
        telescope_id: int
            telescope's id
        Returns float
        """
        return self.lib.get_time_slice(telescope_id)
```

Reply:
OK, it should perform as expected then.